docs/using.md

```diff
@@ -494,6 +541,73 @@ spec:
           ${inputs.resources.testCluster.Name} apply -f /workspace/service.yaml'
 ```
 
+### Storage Resource
+
+Storage resource represents a blob storage, that contains either an object or directory.
+Adding the storage resource as an input to a task will download the blob and allow the task to perform the required actions on the contents of the blob. Blob storage type "Google Cloud Storage"(gcs) is supported as of now.
+
+#### GCS Storage Resource
+
+GCS Storage resource points to "Google Cloud Storage" blob.
+
+To create a GCS type of storage resource using the `PipelineResource` CRD:
+
+```yaml
+apiVersion: pipeline.knative.dev/v1alpha1
+kind: PipelineResource
+metadata:
+  name: wizzbang-storage
+  namespace: default
+spec:
+  type: storage
+  params:
+    - name: type
+      value: gcs
+    - name: location
+      value: gs://some-bucket
+```
+
+Params that can be added are the following:
+
+1. `location`: represents the location of the blob storage.
+1. `type`: represents the type of blob storage. Currently there is implementation for only `gcs`.
+1. `dir`: represents whether the blob storage is a directory or not.
+    1. If artifact is a directory then `-r`(recursive) flag is used to copy all files under source directory to gcs bucket. Eg: `gsutil cp -r source_dir gs://some-bucket`
+    1. If is a single file like zip, tar files then copy will be only 1 level deep.  Eg: `gsutil cp source.tar gs://some-bucket.tar`. This will not trigger copy of sub directories in source directory.
+
+Private buckets can also be configured as storage resources. To access GCS private buckets, service accounts are required with correct permissions. `secrets` field on storage resource is used for configuring this information. Below is an example on how to create a storage resource with service account.
+1. Refer to [official documentation](https://cloud.google.com/compute/docs/access/service-accounts) on how to create service accounts and configuring IAM permissions to access bucket.
+1. Create a kubernetes secret from downloaded service account json key
+
+    ```bash
+    $ kubectl create secret generic bucket-sa --from-file=./service_account.json
+    ```
+
+1. Apply secret to the GCS storage resource in `fieldName` key and `GOOGLE_APPLICATION_CREDENTIALS` value.
+
+    ```yaml
+    apiVersion: pipeline.knative.dev/v1alpha1
+    kind: PipelineResource
+    metadata:
+      name: wizzbang-storage
+      namespace: default
+    spec:
+      type: storage
+      params:
+        - name: type
+          value: gcs
+        - name: url
+          value: gs://some-private-bucket
+        - name: dir
+          value: "directory"
+      secrets:
+        - fieldName: GOOGLE_APPLICATION_CREDENTIALS
+          secretKey: bucket-sa
+          secretName: service_account.json
+    ```
```

Comment:
The use of fieldName here is confusing because this is not actually a field, I would prefer if it was a field on the gcs resource, doesn't have to be part of the generic interface